source/Halibut/Transport/Protocol/MessageExchangeStream.cs

```diff
@@ -335,9 +338,9 @@ void WriteEachStream(IEnumerable<DataStream> streams)
             }
         }
 
-        class MessageEnvelope
+        class MessageEnvelope<T>
         {
-            public object Message { get; set; }
+            public T Message { get; set; }
```

Comment:
Is it worth adding a comment here explaining why it's important for MessageEnvelope to be generic?

Reply:
Done